packages/chakra-components/src/MenuBar/SubMenu.js

```diff
@@ -190,13 +154,24 @@ const SubMenu = ({
     closeOnBlur,
     colorMode,
     mouseOnSubMenuTitle,
+    isCollapsable,
   };
 
   return (
     <SubMenuContext.Provider value={context}>
-      <PseudoBox as="li" role="none" display="flex" {...props}>
+      <PseudoBox
+        as="li"
+        role="none"
+        display="flex"
+        flexDirection={isCollapsable ? "column" : "row"}
```

Comment:
Instead of creating the style prop like this.

Create a separate condition about return like below

```suggestion
let collapsableStyleProps = {}
if(isCollapsable) {
collapsableStyleProps = {flexDirection: "column" }}
}
```
 In Pseudobox after {...styleProps} include this as {...collapsableStyleProps}.

We are doing this so the base styles are overridden properly

Reply:
done